### PR TITLE
Use clang-format 14 instead of 13

### PR DESCRIPTION
**Pull request by DMaroo**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Bump `clang-format` to version 14 instead of 13

**Test plan**

Linter CI should be green

**Closing issues**

None

### CODE REVIEW

File: librz/asm/arch/include/mybfd.h [hunk 1]
```diff
@@ -1506,7 +1506,9 @@ extern const struct bfd_symbol *const bfd_ind_symbol;
 			(struct bfd_symbol **)SYM_PTR, \
 \
 			/* map_head, map_tail                                            */ \
-			{ NULL }, { NULL } \
+			{ NULL }, { \
```
Comment (XVilka) [inline librz/asm/arch/include/mybfd.h]: This one looks weird.

**Review by XVilka — APPROVED** [review]

File: librz/asm/arch/include/mybfd.h (hunk 1 above)
Comment (DMaroo) [reply librz/asm/arch/include/mybfd.h]: I agree. Should I change the `.clang-format`?

Comment (XVilka) [reply librz/asm/arch/include/mybfd.h]: Do you know what would change this behavior? It might be just a one-off case.

Comment (DMaroo) [reply librz/asm/arch/include/mybfd.h]: No, I do not know. I'll have a look at the documentation and the changelog.

Comment (DMaroo) [reply librz/asm/arch/include/mybfd.h]: I am unable to find the option which can make these blocks look as they did in clang-format 13.

Changelog: https://releases.llvm.org/14.0.0/tools/clang/docs/ReleaseNotes.html#clang-format
Documentation: https://clang.llvm.org/docs/ClangFormatStyleOptions.html

Comment (XVilka) [reply librz/asm/arch/include/mybfd.h]: You can just add off/on comments

Comment (XVilka) [reply librz/asm/arch/include/mybfd.h]: Maybe it also makes sense to open a bug in clang-format itself, it does look like it

Comment (DMaroo) [reply librz/asm/arch/include/mybfd.h]: I'll probably ask first to the folks at clang-format, and see if it is really not possible to do this. If it indeed is not possible, we can add off/on directives till clang-format fixes it.

Comment (ret2libc) [reply librz/asm/arch/include/mybfd.h]: i think we can just remove this whole `BFD_FAKE_SECTION` macro as it does not seem to be used anywhere.

**Review by ret2libc — COMMENTED** [review]
LGTM apart from removing that macro.

File: librz/asm/arch/include/mybfd.h (hunk 1 above)
Comment (DMaroo) [reply librz/asm/arch/include/mybfd.h]: @XVilka I did ask in the official LLVM Discord. There has been no response. I guess, we'll just let it be for now. Maybe we can change the `.clang-format` later, if we find the correct option.

@ret2libc Sure, I will remove that macro.

**Review by XVilka — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]

**Pull request merged**